File: Algorithms/487. Max Consecutive Ones II/javascript/solution.brute-force.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function (nums) {
    let longestSequence = 0;
    for (let left = 0; left < nums.length; left++) {
        let numZeroes = 0;
        for (let right = left; right < nums.length; right++) {
            if (numZeroes === 2) {
                break;
            }
            if (nums[right] === 0) {
                numZeroes++;
            }
            if (numZeroes <= 1) {
                longestSequence = Math.max(longestSequence, right - left + 1);
            }
        }
    }
    return longestSequence;
};

File: Algorithms/487. Max Consecutive Ones II/swift/solution.brute-force.swift
class Solution {
    func findMaxConsecutiveOnes(_ nums: [Int]) -> Int {
        var longestSequence: Int = 0

        for left in 0..<nums.count {
            var numZeroes: Int = 0

            for right in left..<nums.count {
                if numZeroes == 2 {
                    break
                }

                if nums[right] == 0 {
                    numZeroes += 1
                }
                if numZeroes <= 1 {
                    longestSequence = max(longestSequence, right - left + 1)
                }
            }
        }

        return longestSequence
    }
}
